fdbrpc/FlowTransport.actor.cpp

```diff
@@ -1584,10 +1585,15 @@ static ReliablePacket* sendPacket(TransportData* self,
 
 	// Reserve some space for packet length and checksum, write them after serializing data
 	SplitBuffer packetInfoBuffer;
-	uint32_t len, checksum = 0;
+	uint32_t len;
+	XXH64_hash_t checksum = 0;
+	XXH3_state_t* checksumState = nullptr;
+
 	int packetInfoSize = PACKET_LEN_WIDTH;
 	if (checksumEnabled) {
 		packetInfoSize += sizeof(checksum);
+		checksumState = XXH3_createState();
+		ASSERT(XXH3_64bits_reset(checksumState) == XXH_OK);
```

Comment:
I don't think using `ASSERT` here is a good idea. In general, I don't think we should ever have an expression with side-effects in an assertion...

Basically in the far future someone might think it is a good idea to redefine `ASSERT` to do nothing and that would cause behavior changes. It's even possible to disable assertions through the command line...

So what I am saying is, these need to be replaced with something like this:

```suggestion
		if (XXH3_64bits_reset(checksumState) != XXH_OK) {
			throw internal_error();
		}
```